client/recoil.lua

```diff
@@ -1,156 +1,55 @@
-
-	-- Shotguns
-	[487013001] = 0.4,			--['weapon_pumpshotgun']
-	[2017895192] = 0.7,			--['weapon_sawnoffshotgun']
-	[-494615257] = 0.4,			--['weapon_assaultshotgun']
-	[-1654528753] = 0.2,		--['weapon_bullpupshotgun']
-	[-1466123874] = 0.7,		--['weapon_musket']
-	[984333226] = 0.2,			--['weapon_heavyshotgun']
-	[-275439685] = 0.7,			--['weapon_dbshotgun']
-	[317205821] = 0.2,			--['weapon_autoshotgun']
-	[1432025498] = 0.4,			--['weapon_pumpshotgun_mk2']
-	[94989220] = 0.0,			--['weapon_combatshotgun']
-
-	-- Assault Rifles
-	[-1074790547] = 0.5,		--['weapon_assaultrifle']
-	[961495388] = 0.2,			--['weapon_assaultrifle_mk2']
-	[-2084633992] = 0.3,		--['weapon_carbinerifle']
-	[-86904375] = 0.1,			--['weapon_carbinerifle_mk2']
-	[-1357824103] = 0.1,		--['weapon_advancedrifle']
-	[-1063057011] = 0.2,		--['weapon_specialcarbine']
-	[2132975508] = 0.2,			--['weapon_bullpuprifle']
-	[1649403952] = 0.3,			--['weapon_compactrifle']
-	[-1768145561] = 0.2,		--['weapon_specialcarbine_mk2']
-	[-2066285827] = 0.2,		--['weapon_bullpuprifle_mk2']
-	[-1658906650] = 0.0,		--['weapon_militaryrifle']
-
-	-- Light Machine Guns
-	[-1660422300] = 0.1,		--['weapon_mg']
-	[2144741730] = 0.1,			--['weapon_combatmg']
-	[1627465347] = 0.1,			--['weapon_gusenberg']
-	[-608341376] = 0.1,			--['weapon_combatmg_mk2']
-
-	-- Sniper Rifles
-	[100416529] = 0.5,			--['weapon_sniperrifle']
-	[205991906] = 0.7,			--['weapon_heavysniper']
-	[-952879014] = 0.3,			--['weapon_marksmanrifle']
-	[856002082] = 1.2,			--['weapon_remotesniper']
-	[177293209] = 0.6,			--['weapon_heavysniper_mk2']
-	[1785463520] = 0.3,			--['weapon_marksmanrifle_mk2']
-
-	-- Heavy Weapons
-	[-1312131151] = 0.0,		--['weapon_rpg']
-	[-1568386805] = 1.0,		--['weapon_grenadelauncher']
-	[1305664598] = 1.0,			--['weapon_grenadelauncher_smoke']
-	[1119849093] = 0.1,			--['weapon_minigun']
-	[2138347493] = 0.3,			--['weapon_firework']
-	[1834241177] = 2.4,			--['weapon_railgun']
-	[1672152130] = 0.0,			--['weapon_hominglauncher']
-	[125959754] = 0.5,			--['weapon_compactlauncher']
-	[-1238556825] = 0.3,		--['weapon_rayminigun']
-
-	-- Throwables
-	-- [-1813897027] = 0.3,		--['weapon_grenade']
-	-- [-1600701090] = 0.3,		--['weapon_bzgas']
-	-- [615608432] = 0.3,		--['weapon_molotov']
-	-- [741814745] = 0.3,		--['weapon_stickybomb']
-	-- [-1420407917] = 0.3,		--['weapon_proxmine']
-	-- [126349499] = 0.3,		--['weapon_snowball']
-	-- [-1169823560] = 0.3,		--['weapon_pipebomb']
-	-- [600439132] = 0.3,		--['weapon_ball']
-	-- [-37975472] = 0.3,		--['weapon_smokegrenade']
-	-- [1233104067] = 0.3,		--['weapon_flare']
-
-	-- Miscellaneous
-	-- [883325847] = 0.3,		--['weapon_petrolcan']
-	-- [101631238] = 0.3,		--['weapon_fireextinguisher']
-	-- [-1168940174] = 0.3,		--['weapon_hazardcan']
-}
+lib.onCache('weapon', function(weapon)
+	if weapon then
+		if weapons[weapon] and weapons[weapon].disableHeadshot then
+			-- i dunno how this native work
+			SetPedSuffersCriticalHits(cache.ped, false)
```

Comment:
How will this even work?

If you have a specific weapon you no longer die from taking headshots? thats kinda wierd